src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,147 @@
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registerUser_WithLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidLoginLength_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("valid123");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithPasswordNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword(null);
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidPasswordLength_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("qwert");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAgeNull_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithInvalidAge_NotOk() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllInvalidFields_NotOk() {
+        user.setLogin("Steve");
+        user.setPassword("qwert");
+        user.setAge(15);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void registerUser_WithAllValidFields_Ok() {
+        user.setLogin("Steve123");
+        user.setPassword("valid123");
+        user.setAge(20);
+        registrationService.register(user);
+        assertEquals(user, storageDao.get("Steve123"));
+    }
+
+    @Test
+    void getUser_ByLogin_Ok() {
```

Comment:
You are testing register method in this Test class, so remove getUser_ByLogin_Ok() and getUser_ByNotExistLogin_NotOk()